src/Http/Kernel.php

```diff
@@ -0,0 +1,178 @@
+            // it means that document root is the GLPI root directory, but a rewrite rule redirects the request to the PHP router.
+            // This case happen when redirection to PHP router is made by an `.htaccess` file placed in the GLPI root directory,
+            // and has to be handled to support shared hosting where it is not possible to change the web server root directory.
+            $uri_prefix = '';
+        } else {
+            // `$request->server->get('SCRIPT_NAME')` corresponds to the script path relative to server document root.
+            // -> if server document root is `/public`, then `$request->server->get('SCRIPT_NAME')` will be equal to `/index.php`
+            // -> if script is located into a `/glpi-alias` alias directory, then `$request->server->get('SCRIPT_NAME')` will be equal to `/glpi-alias/index.php`
+            $uri_prefix = rtrim(str_replace('\\', '/', dirname($request->server->get('SCRIPT_NAME'))), '/');
+        }
+
+        // Get URI path relative to GLPI (i.e. without alias directory prefix).
+        $path = preg_replace(
+            '/^' . preg_quote($uri_prefix, '/') . '/',
+            '',
+            parse_url($request->server->get('REQUEST_URI') ?? '/', PHP_URL_PATH)
+        );
+
+        require_once $glpi_root . '/src/Http/ProxyRouter.php';
+
+        $proxy = new \Glpi\Http\ProxyRouter($glpi_root, $path);
+        $proxy->handleRedirects($uri_prefix);
+
+        if ($proxy->isTargetAPhpScript() && $proxy->isPathAllowed() && ($target_file = $proxy->getTargetFile()) !== null) {
+            // Ensure `getcwd()` and inclusion path is based on requested file FS location.
+            chdir(dirname($target_file));
+
+            // (legacy) Redefine some $_SERVER variables to have same values whenever scripts are called directly
+            // or through current router.
+            $target_path     = $uri_prefix . $proxy->getTargetPath();
+            $target_pathinfo = $proxy->getTargetPathInfo();
+            $_SERVER['PATH_INFO']       = $target_pathinfo;
+            $_SERVER['PHP_SELF']        = $target_path;
+            $_SERVER['SCRIPT_FILENAME'] = $target_file;
+            $_SERVER['SCRIPT_NAME']     = $target_path;
+
+            // New server overrides:
+            $request->server->set('PATH_INFO', $target_pathinfo);
+            $request->server->set('PHP_SELF', $target_path);
+            $request->server->set('SCRIPT_FILENAME', $target_file);
+            $request->server->set('SCRIPT_NAME', $target_path);
+
+            // Execute target script.
+            $this->requireFile($target_file, $request);
+            exit;
+        }
+
+        $proxy->proxify();
+
+        if (str_contains(implode(";", headers_list()), ';Content-type:')) {
+            // "proxify()" sends headers for files, and if it does, we must exit() to avoid running Symfony's router
+            exit;
+        }
```

Comment:
IMHO, it would be preferable to return either a boolean, either `Response|null` in the `ProxyRouter::proxify()` method, instead of doing this kind of check that may be impacted by an unexpected behaviour in a plugin.

Reply:
I specifically did this to avoid having to change the ProxyRouter and keep the initial behavior for now.

Once the entire routing thing is rewritten, the legacy system could be a simple kernel.controller event, for instance.